app/src/main/java/com/github/sdp/ratemyepfl/activity/ReviewableListActivity.kt

```diff
@@ -52,17 +54,15 @@ abstract class ReviewableListActivity<T : Reviewable> : AppCompatActivity() {
         }
     }
 
-    abstract fun getExtraString(): String
-
-    abstract fun getExtraClass(): Class<ReviewsListActivity<T>>
-
     abstract fun getMenuString(): Int
 
     abstract fun getSearchViewString(): Int
 
-    private fun displayReviews(t: T) {
-        val intent = Intent(this, getExtraClass())
-        intent.putExtra(getExtraString(), t.toJSON())
+    abstract fun getReviewClass(): Class<ReviewActivity>
+
+    fun displayReviews(t: T){
+        val intent = Intent(this, getReviewClass())
+        intent.putExtra(AddReviewActivity.EXTRA_ITEM_REVIEWED, t.id)
```

Comment:
Yes, I got rid of the serialization in this part, because for the info page we will need to fetch the reviewable from the DB to get the most recent version.